### PR TITLE
governance: shorten README based on template changes

### PR DESCRIPTION
**Pull request by philips**
With the template updates in https://github.com/kubernetes/community/commit/32d184cee2ecb7c72f9de75e687f4a88f7c36413#diff-a068083f7bbb2cb58e8c350f4752056d we don't need as much explanation of these docs. Remove extra sentences and focus the goal.

### CODE REVIEW

File: committee-steering/governance/README.md [hunk 1]
```diff
@@ -30,45 +30,19 @@ All Kubernetes SIGs must define a charter defining the scope and governance of t
 
 ## How to use the templates
 
-When developing or modifying a SIG governance doc, the intention is for SIGs to use the templates (*under development*)
-as a common set of options SIGs may choose to incorporate into their own governance structure.  It is recommended that
-SIGs start by looking at the [Recommendations and requirements] for SIG governance docs and consider what structure
-they think will work best for them before pulling items from the templates.
+SIGs should use [the template][Short Template] as a starting point. This document links to the recommended SIG Governance][sig-governance] but SIGs may optionally record deviations from these defaults in their charter.
```
Comment (michelleN) [inline committee-steering/governance/README.md]: missing a `[` before SIG in this line

**Comment by pwittrock** [comment]
LGTM after address @michelleN 's comment

**Comment by philips** [comment]
@spiffxp @michelleN Addressed feedback. Ready for that LGTM :)

**Pull request merged**